catroid/src/test/java/org/catrobat/catroid/test/formulaeditor/ParserTestFunctions.java

```diff
@@ -399,6 +399,49 @@ public void testArctan() {
 		FormulaEditorTestUtil.testSingleParameterFunction(Functions.ARCTAN, firstParameterList, Double.NaN, testSprite);
 	}
 
+	@Test
+	public void testArctan2() {
+		List<InternToken> firstParameter = new LinkedList<InternToken>();
+		List<InternToken> secondParameter = new LinkedList<InternToken>();
+		firstParameter.add(new InternToken(InternTokenType.OPERATOR, Operators.MINUS.name()));
+		firstParameter.add(new InternToken(InternTokenType.NUMBER, "1"));
+		secondParameter.add(new InternToken(InternTokenType.NUMBER, "0"));
+
+		FormulaEditorTestUtil.testDoubleParameterFunction(Functions.ARCTAN2, firstParameter,
+				secondParameter, -90.0, testSprite);
+
+		firstParameter = new LinkedList<InternToken>();
+		secondParameter = new LinkedList<InternToken>();
+		firstParameter.add(new InternToken(InternTokenType.OPERATOR, Operators.MINUS.name()));
+		firstParameter.add(new InternToken(InternTokenType.NUMBER, "1"));
+		secondParameter.add(new InternToken(InternTokenType.OPERATOR, Operators.MINUS.name()));
+		secondParameter.add(new InternToken(InternTokenType.NUMBER, "1"));
+
+		FormulaEditorTestUtil.testDoubleParameterFunction(Functions.ARCTAN2, firstParameter,
+				secondParameter, -135.0, testSprite);
+
+		List<InternToken> firstParameterList = FormulaEditorTestUtil.buildBinaryOperator(InternTokenType.NUMBER, "5", Operators.PLUS,
+				InternTokenType.STRING, "datString");
+		List<InternToken> secondParameterList = FormulaEditorTestUtil.buildBinaryOperator(InternTokenType.NUMBER, "5", Operators.MULT,
+				InternTokenType.STRING, "anotherString");
+		FormulaEditorTestUtil.testDoubleParameterFunction(Functions.ARCTAN2, firstParameterList, secondParameterList,
```

Comment:
that is a lot of asserts with different formulas for one test. @84n4n4 has been trying to get people to write smaller, more concise tests. his rationale is quite reasonable IMO, because with more, smaller tests you, as a developer immediately see what's wrong and in the test overview it more realistically represents what's broken and what works. e.g. if the arctan formula suddenly breaks if you put strings into it, then the whole test fails, but you don't know it was because of the strings and you have to find out if it was because of numbers, strings or something else. besides, there is one red test that says arctan not working, but if you split it you can easily see what's wrong and you see that e.g. arc tan with strings is broken, but works fine with numbers. - I guess you get the idea.